cli/spcli/sectors.go

```diff
@@ -0,0 +1,1395 @@
+	"github.com/filecoin-project/lotus/lib/tablewriter"
+)
+
+type OnDiskInfoGetter func(cctx *cli.Context, id abi.SectorNumber, onChainInfo bool) (api.SectorInfo, error)
+
+func SectorsStatusCmd(getActorAddress ActorAddressGetter, getOnDiskInfo OnDiskInfoGetter) *cli.Command {
+	return &cli.Command{
+		Name:      "status",
+		Usage:     "Get the seal status of a sector by its number",
+		ArgsUsage: "<sectorNum>",
+		Flags: []cli.Flag{
+			&cli.BoolFlag{
+				Name:    "log",
+				Usage:   "display event log",
+				Aliases: []string{"l"},
+			},
+			&cli.BoolFlag{
+				Name:    "on-chain-info",
+				Usage:   "show sector on chain info",
+				Aliases: []string{"c"},
+			},
+			&cli.BoolFlag{
+				Name:    "partition-info",
+				Usage:   "show partition related info",
+				Aliases: []string{"p"},
+			},
+			&cli.BoolFlag{
+				Name:  "proof",
+				Usage: "print snark proof bytes as hex",
+			},
+		},
+		Action: func(cctx *cli.Context) error {
+			ctx := lcli.ReqContext(cctx)
+
+			if cctx.NArg() != 1 {
+				return lcli.IncorrectNumArgs(cctx)
+			}
+
+			id, err := strconv.ParseUint(cctx.Args().First(), 10, 64)
+			if err != nil {
+				return err
+			}
+
+			onChainInfo := cctx.Bool("on-chain-info")
+
+			var status api.SectorInfo
+			if getOnDiskInfo != nil {
+				status, err = getOnDiskInfo(cctx, abi.SectorNumber(id), onChainInfo)
+				if err != nil {
+					return err
+				}
+				fmt.Printf("SectorID:\t%d\n", status.SectorID)
+				fmt.Printf("Status:\t\t%s\n", status.State)
+				fmt.Printf("CIDcommD:\t%s\n", status.CommD)
+				fmt.Printf("CIDcommR:\t%s\n", status.CommR)
+				fmt.Printf("Ticket:\t\t%x\n", status.Ticket.Value)
+				fmt.Printf("TicketH:\t%d\n", status.Ticket.Epoch)
+				fmt.Printf("Seed:\t\t%x\n", status.Seed.Value)
+				fmt.Printf("SeedH:\t\t%d\n", status.Seed.Epoch)
+				fmt.Printf("Precommit:\t%s\n", status.PreCommitMsg)
+				fmt.Printf("Commit:\t\t%s\n", status.CommitMsg)
+				if cctx.Bool("proof") {
+					fmt.Printf("Proof:\t\t%x\n", status.Proof)
+				}
+				fmt.Printf("Deals:\t\t%v\n", status.Deals)
+				fmt.Printf("Retries:\t%d\n", status.Retries)
+				if status.LastErr != "" {
+					fmt.Printf("Last Error:\t\t%s\n", status.LastErr)
+				}
+
+				fmt.Printf("\nExpiration Info\n")
+				fmt.Printf("OnTime:\t\t%v\n", status.OnTime)
+				fmt.Printf("Early:\t\t%v\n", status.Early)
+
+			} else {
+				onChainInfo = true
+			}
+
+			maddr, err := getActorAddress(cctx)
+			if err != nil {
+				return err
+			}
+
+			if onChainInfo {
+				fullApi, closer, err := lcli.GetFullNodeAPI(cctx)
+				if err != nil {
+					return err
+				}
+				defer closer()
+
+				head, err := fullApi.ChainHead(ctx)
+				if err != nil {
+					return xerrors.Errorf("getting chain head: %w", err)
+				}
+
+				status, err := fullApi.StateSectorGetInfo(ctx, maddr, abi.SectorNumber(id), head.Key())
+				if err != nil {
+					return err
+				}
+				fmt.Printf("\nSector On Chain Info\n")
+				fmt.Printf("SealProof:\t\t%x\n", status.SealProof)
```

Comment:
The on-chain info should have quite a bit more info that we can put in here. At least we can get the SectorID (that is just sector number + miner ID) and CommR + probably some more things

Reply:
This is unchanged from what was here before. Feel free to add what you would like. 
I added something like what you described for SectorID, but i don't think it's what you want, because I could not find an example of how those should appear in the output.